src/ui/server/controllers/insurance/your-buyer/failed-to-pay-on-time/index.test.ts

```diff
@@ -0,0 +1,283 @@
+    FAILED_TO_PAY_CHECK_AND_CHANGE,
+  },
+  CHECK_YOUR_ANSWERS: { YOUR_BUYER: CHECK_AND_CHANGE_ROUTE },
+  PROBLEM_WITH_SERVICE,
+} = INSURANCE_ROUTES;
+
+const { FAILED_PAYMENTS } = BUYER_FIELD_IDS;
+
+describe('controllers/insurance/your-buyer/failed-to-pay-on-time', () => {
+  let req: Request;
+  let res: Response;
+
+  beforeEach(() => {
+    req = mockReq();
+    res = mockRes();
+  });
+
+  afterAll(() => {
+    jest.resetAllMocks();
+  });
+
+  describe('pageVariables', () => {
+    it('should have correct properties', () => {
+      const expected = {
+        FIELD_ID,
+        PAGE_CONTENT_STRINGS,
+        SAVE_AND_BACK_URL: `${INSURANCE_ROOT}/${referenceNumber}${SAVE_AND_BACK}`,
+        FIELD_HINT: FIELDS[FIELD_ID].HINT,
+      };
+
+      expect(pageVariables(referenceNumber)).toEqual(expected);
+    });
+  });
+
+  describe('FIELD_ID', () => {
+    it('should have the correct FIELD_ID', () => {
+      expect(FIELD_ID).toEqual(FAILED_PAYMENTS);
+    });
+  });
+
+  describe('TEMPLATE', () => {
+    it('should have the correct template defined', () => {
+      expect(TEMPLATE).toEqual(TEMPLATES.SHARED_PAGES.SINGLE_RADIO);
+    });
+  });
+
+  describe('HTML_FLAGS', () => {
+    it('should have correct properties', () => {
+      const expected = {
+        HORIZONTAL_RADIOS: true,
+        NO_RADIO_AS_FIRST_OPTION: true,
+      };
+
+      expect(HTML_FLAGS).toEqual(expected);
+    });
+  });
+
+  describe('PAGE_CONTENT_STRINGS', () => {
+    it('should have the correct strings', () => {
+      expect(PAGE_CONTENT_STRINGS).toEqual(PAGES.INSURANCE.YOUR_BUYER.FAILED_PAYMENTS);
+    });
+  });
+
+  describe('get', () => {
+    it('should render template', async () => {
+      await get(req, res);
+
+      const expectedVariables = {
+        ...singleInputPageVariables({
+          ...pageVariables(referenceNumber),
+          BACK_LINK: req.headers.referer,
+          HTML_FLAGS,
+        }),
+        ...pageVariables(referenceNumber),
+        userName: getUserNameFromSession(req.session.user),
+        applicationAnswer: mockApplication.buyer.buyerTradingHistory[FIELD_ID],
+      };
+
+      expect(res.render).toHaveBeenCalledWith(TEMPLATE, expectedVariables);
+    });
+
+    describe('when there is no application', () => {
+      beforeEach(() => {
+        delete res.locals.application;
+      });
+
+      it(`should redirect to ${PROBLEM_WITH_SERVICE}`, () => {
+        get(req, res);
+
+        expect(res.redirect).toHaveBeenCalledWith(PROBLEM_WITH_SERVICE);
+      });
+    });
+  });
+
+  describe('post', () => {
+    const validBody = {
+      [FIELD_ID]: 'true',
+    };
+
+    beforeEach(() => {
+      mapAndSave.buyerTradingHistory = jest.fn(() => Promise.resolve(true));
+    });
+
+    describe('when there are no validation errors', () => {
+      beforeEach(() => {
+        req.body = validBody;
+      });
+
+      it(`should redirect to the next page the totalContractValueOverThreshold is not true`, async () => {
```

Comment:
Do we have tests for when `totalContractValueOverThreshold` is true?

Reply:
yes, line 198

Reply:
OK great, can we just add a "when" to this test description please